v0/destinations/kustomer/transform.js

```diff
@@ -0,0 +1,194 @@
+    getFieldValueFromMessage(message, "lastName")
+  ) {
+    kustomerPayload.name = `${getFieldValueFromMessage(
+      message,
+      "firstName"
+    )} ${getFieldValueFromMessage(message, "lastName")}`;
+  }
+
+  if (getFieldValueFromMessage(message, "email")) {
+    kustomerPayload.emails = kustomerPayload.emails
+      ? kustomerPayload.emails
+      : [{ type: "home", email: getFieldValueFromMessage(message, "email") }];
+  }
+
+  if (getFieldValueFromMessage(message, "phone")) {
+    kustomerPayload.phones = kustomerPayload.phones
+      ? kustomerPayload.phones
+      : [{ type: "home", phone: getFieldValueFromMessage(message, "phone") }];
+  }
+  const url = get(message, "traits.website")
+    ? get(message, "traits.website")
+    : get(message, "context.traits.website");
+  if (url) {
+    kustomerPayload.urls = [
+      {
+        url
+      }
+    ];
+  }
+
+  const address = getFieldValueFromMessage(message, "address");
+  if (address) {
+    kustomerPayload.locations = [
+      {
+        type: "home",
+        address:
+          typeof address === "string"
+            ? address
+            : `${address.street}, ${address.city}, ${address.state}, ${address.postalCode}`
+      }
+    ];
+  }
+
+  removeUndefinedAndNullValues(kustomerPayload);
+  return kustomerPayload;
+};
+
+// Main process function responsible for building payload for all
+// type of events.
+const responseBuilderSimple = async (message, category, destination) => {
+  let payload = {};
+  let targetUrl = BASE_ENDPOINT;
+  let userExists = false;
+  // In case of identify type of event first extract the anonymousId, userId
+  // and search if any Kustomer is present in destination.
+  // If present update the same kustomer with the given payload.
+  // else create a new kustomer.
+  //-------------------------------------------------------------
+  // Get Kustomer: https://apidocs.kustomer.com/#ff41f372-6144-4c64-9712-662ee5ef1c33
+  // Create Kustomer: https://apidocs.kustomer.com/#07bd1072-4d4b-4875-b526-8369d711e811
+  // Update Kustomer: https://apidocs.kustomer.com/#077d653a-184e-4153-8133-d24b6427c1ae
+  if (message.type.toLowerCase() == EventType.IDENTIFY) {
+    const idArr = [
+      get(message, "anonymousId"),
+      getFieldValueFromMessage(message, "userIdOnly")
+    ];
+    let response;
+    await Promise.all(
+      idArr.map(async id => {
+        try {
+          response = await axios.get(
+            `${BASE_ENDPOINT}/v1/customers/externalId=${id}`,
+            {
+              headers: {
+                Authorization: `Bearer ${destination.Config.apiKey}`
+              }
+            }
+          );
+        } catch (err) {
+          logger.debug("Error while fetching customer info");
+        }
+        // If Kustomer Exists url to user : Update kustomer
+        if (response && response.status === 200) {
+          userExists = true;
+          targetUrl = `${targetUrl}/v1/customers/${response.data.data.id}?replace=false`;
```

Comment:
Also, let's add support from `externalId` so that we can skip the API call from here if `kustomerId` is supplied under `context.externalId`

Reply:
Not sure if that will be preferable, as we are primarily mapping userId (or anonId) as externalId (which is similar behaviour for other platform). Maybe we can discuss this point for clarity as I might be missing something

Reply:
The idea of `externalId` is to let the users supply the essential id themselves. This is a very useful feature for the WH actions users. Ideally, they'll have the `kustomerId` in the WH and they can pass it on to the downstream flow so that we don't have to look-up using `email/userId/anonymousId`.